### PR TITLE
clipboard v1

### PR DESCRIPTION
**Pull request by LeonardPoissant**

### CODE REVIEW

**Review by aaspinwall — CHANGES_REQUESTED** [review]
Great! Can you please showcase it on a page?
it could be /clipboard so it doesn't conflict with other work

**Comment by LeonardPoissant** [comment]
it needs relative info from a parent component.I'm not sure how to not make conflicts to showcase it. I'll work on that. Here's a gif in the mean time : https://i.gyazo.com/32a164bf66418d7528198d09210c755d.gif

**Comment by andrewbaldwin44** [comment]
JavaScript has [https://developer.mozilla.org/en-US/docs/Web/API/Document/execCommand](execCommand) which could be used instead of another dependency.

I would say we don't usually want to bring in 100's of lines of code for something we're only using once right?

**Comment by aaspinwall** [comment]
Agree, anything that we can roll ourselves works better

**Comment by aaspinwall** [comment]
> it needs relative info from a parent component.I'm not sure how to not make conflicts to showcase it. I'll work on that. Here's a gif in the mean time : https://i.gyazo.com/32a164bf66418d7528198d09210c755d.gif

It works great! Just make a new file in the pages folder and add it. That's what I mean by showcasing it

**Comment by LeonardPoissant** [comment]
I don't mind changing it, but we spoke about this. I have no clue what the warning about execcommand  in the literature is about and so I chose to go with a package that seemed to be dealing with that.

**Comment by aaspinwall** [comment]
Don't worry about it. The package is fine 🦝

**Comment by andrewbaldwin44** [comment]
> I don't mind changing it, but we spoke about this. I have no clue what the warning about execommand in the literature is about and so I chose to go with a package that seemed to be dealing with that. I'll be more thorough next time I bring something up.

Yah that's ok 😃 the only reason I bring it up is because there's always a lot to consider when bringing in a new dependency (bundle size, whether the dependency is being maintained, etc.).

If you ever encounter an error that you don't know how to deal with you can always post it in the channel or ping one of us! I don't see `execCommand` raising any issues. Make sure you have `React` in-scope at the top of your file and use `createRef` to get a reference to the input field and that should do it :)

File: frontend/components/copytoclipboard.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
+import { useState }  from "react";
```
Comment (andrewbaldwin44) [inline frontend/components/copytoclipboard.js:1]: import React, { useState }  from "react";

Comment (ShaiGilboa) [reply frontend/components/copytoclipboard.js:1]: I think the new React doesn't require to be imported :)

**Comment by ShaiGilboa** [comment]
@andrewbaldwin44 
screenshot (2)

**Comment by andrewbaldwin44** [comment]
> @andrewbaldwin44
> screenshot (2)

Ah well clearly I should have done my research. What about the `Clipboard API`? https://developer.mozilla.org/en-US/docs/Web/API/Clipboard_API. Currently only missing support in IE.

**Comment by ShaiGilboa** [comment]
Seems legit- I recommend we merge the current version Leonard made, and open an issue (for future self) to try and re-write it without the dep

**Comment by LeonardPoissant** [comment]
We have 3 options I think. go with execCommand(which was created for IE apparently and is as you all saw going obsolete...so long IE), use the clipboard_Api(apparently behaves well on multiple browsers), or use a third party library.

**Comment by ShaiGilboa** [comment]
Agreed. I'm don't have strong feelings one way or another - But I do think we should have a bit of error handling, for example there is a "result" Boolean that we can use to check that the copy action was made successfully 
screenshot (3)
Or if we use the API then have a `.catch`

Also - I think the integrated component should have a <p> or a <hX> tag that shows the text that is going to be copied to the clipboard (so people will know what is being copied, and if auto-copy-tp-clipboard doesn't work, they can copy manually )

**Comment by LeonardPoissant** [comment]
perhaps we should set community guide lines/procedures for adding third party libraries. This is just a tiny component lol.

**Comment by LeonardPoissant** [comment]
Talked with Andrew earlier and I think we can go without a library. If everyone is all right with I'll use the newest stuff, clipboard_api. Also as per Andrew's suggestion, I will add visual cues. End result would look something like:

copy invite link: "some invite link" copytocb icon

**Review by aaspinwall — COMMENTED** [review]
Looks good! I'll fix the conflicts with the new packages today

File: frontend/components/ui/copytoclipboard.jsx [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+//TODO: finetune styling. 
+
+const CopyToClipBoardButton =({text})=>{
+    const [copy, setCopy] = useState(false);
+    const inputRef =useRef(null);
+
+    const handleClick = async ()=>{
+      //if the clipboard_api is not supported we use the ol' excCommand to copy to clipboard.
+      if(!navigator.clipboard){
```
Comment (andrewbaldwin44) [inline frontend/components/ui/copytoclipboard.jsx:23]: Ooo that's a smart way to do it! Nice! :)

**Comment by andrewbaldwin44** [comment]
I tested it on Chrome and Firefox and it works a treat! Nicely done Leonard! 💯 

And I like the snazzy copy button haha

**Comment by LeonardPoissant** [comment]
thanks. I'm thinking we could have another icon instead of the boring clipboard one. Also, I'd like to proof this function a bit more, I'm not sure my if statement would go through.

**Comment by aaspinwall** [comment]
> thanks. I'm thinking we could have another icon instead of the clipboard one. Also, I'd like to proof this function a bit more, I'm not sure my if statement would go through.

Sure, I could re open it for you. It could take an icon as props. Deciding on the icons is beyond this stage but it would be nice to have some modularity

**Comment by LeonardPoissant** [comment]
precisely, let's talk fe on wednesday!

**Pull request merged**